applications/GeoMechanicsApplication/custom_workflows/dgeoflow.cpp

```diff
@@ -569,32 +569,15 @@ namespace Kratos
             for (Element element : model_part.Elements())
             {
                 auto rGeom = element.GetGeometry();
-                auto rProp = element.GetProperties();
+                const auto rProp = element.GetProperties();
                 
-                for (unsigned int node = 0; node < 3; ++node)
-                {
-                    array_1d<double, 3> NodeVolumeAcceleration;
-                    noalias(NodeVolumeAcceleration) = rGeom[node].FastGetSolutionStepValue(VOLUME_ACCELERATION, 0);
-                    const double g = norm_2(NodeVolumeAcceleration);
-                    if (g > std::numeric_limits<double>::epsilon())
-                    {
-                        const double FluidWeight = g * rProp[DENSITY_WATER];
+                const auto NodalHydraulicHead = GeoElementUtilities::CalculateNodalHydraulicHeadFromWaterPressures<3>(rGeom, rProp);
 
-                        array_1d<double, 3> NodeCoordinates;
-                        noalias(NodeCoordinates) = rGeom[node].Coordinates();
-                        array_1d<double, 3> NodeVolumeAccelerationUnitVector;
-                        noalias(NodeVolumeAccelerationUnitVector) = NodeVolumeAcceleration / g;
-
-                        const double WaterPressure = rGeom[node].FastGetSolutionStepValue(WATER_PRESSURE);
-                        rGeom[node].SetValue(*element_var, -inner_prod(NodeCoordinates, NodeVolumeAccelerationUnitVector) - PORE_PRESSURE_SIGN_FACTOR * WaterPressure / FluidWeight);
-                    }
-                    else
-                    {
-                        rGeom[node].SetValue(*element_var, 0.0);
-                    }
+            	for (unsigned int node = 0; node < 3; ++node)
+                {
+                    rGeom[node].SetValue(*element_var, NodalHydraulicHead[node]);
```

Comment:
A minor thing: if you convert `element_var` from a pointer to a reference-to-const, you no longer need to dereference it in the remainder of the function. Since `KratosComponents<Variable<double>>::Get("HYDRAULIC_HEAD")` returns `const Kratos::Variable<double>&`, you may simply write
```
const auto& element_var = KratosComponents<Variable<double>>::Get("HYDRAULIC_HEAD");
```
And where you use `element_var`, you'd remove the dereference operator (`*`).

Reply:
Done